test/androidTest/java/org/thoughtcrime/securesms/PinnedMessageEspressoTest.java

```diff
@@ -4,34 +4,67 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import android.content.Intent;
 import android.support.test.runner.AndroidJUnit4;
 import android.support.test.rule.ActivityTestRule;
 import android.support.test.filters.LargeTest;
+
 import static android.support.test.espresso.Espresso.onView;
-import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
-import static android.support.test.espresso.assertion.ViewAssertions.matches;
 
 @RunWith(AndroidJUnit4.class)
 @LargeTest
 public class PinnedMessageEspressoTest {
 
     @Rule
-    public ActivityTestRule<PinnedMessageActivity> pinnedActivityRule =
-            new ActivityTestRule(PinnedMessageActivity.class, true, false);
+    public ActivityTestRule<ConversationListActivity> mainActivityRule =
+            new ActivityTestRule(ConversationListActivity.class, true, false);
 
     @Test
     public void pageExists() {
-        Intent intent = new Intent();
+        EspressoHelper helper = new EspressoHelper(mainActivityRule);
+
+        helper
+            .createConversation()
+            .goPinned()
+            .done();
```

Comment:
I like this pattern, it is clear what you are trying to do. however, I believe out of this we can create a small ui testing framework for us by pull up the common methods such as "sendMessage()" to a parent class and then the children classes such as EspressoHelperPinMessage class can add its own functionalities

Reply:
Yep that is exactly what I was planning on doing